qiskit/qasm2/export.py

```diff
@@ -165,25 +167,53 @@ def dumps(circuit: QuantumCircuit, /) -> str:
         f"{'qreg' if isinstance(reg, QuantumRegister) else 'creg'} {name}[{reg.size}];"
         for name, reg in register_escaped_names.items()
     )
-    instruction_calls = []
-    for instruction in circuit._data:
-        operation = instruction.operation
-        if operation.name == "measure":
+
+    def qasm_line(instruction) -> str | None:
+        if instruction.is_control_flow():
+            operation = instruction.operation
+            if operation.name != "if_else":
+                raise QASM2ExportError("OpenQASM 2 does not support control-flow constructs")
+            if isinstance(operation.condition, expr.Expr) or isinstance(
+                operation.condition[0], Clbit
+            ):
+                # This could be relaxed (somewhat) in the future by attempting to resolve the `Expr`
+                # into a simple "register-equality" form; `Expr` is a superset of allowed conditions
+                raise QASM2ExportError("OpenQASM 2 only supports register-equality conditions")
+            if len(operation.blocks) != 1:
+                raise QASM2ExportError("OpenQASM 2 does not support 'else' statements")
+            body = operation.blocks[0]
+            if len(body.data) != 1:
+                # This could be relaxed in the future by "unrolling" the block and putting
+                # individual `if` conditions on each instruction.
+                raise QASM2ExportError(
```

Comment:
It's stretching my memory, but were we previously able to do c_if on `InstructionSet`s (i.e. multiple instructions) or do we have the same functionality as before if we only allow a single instruction?